internal/extproc/translator/translator.go

```diff
@@ -28,11 +28,8 @@ func isGoodStatusCode(code int) bool {
 	return code >= 200 && code < 300
 }
 
-// Factory creates a [Translator] for the given API schema combination and request path.
-//
-//   - `path`: the path of the request.
-//   - `l`: the logger.
-type Factory func(path string) (Translator, error)
+// Factory creates a [Translator] for the given API schema combination.
+type Factory func() Translator
 
 // NewFactory returns a callback function that creates a translator for the given API schema combination.
 func NewFactory(in, out filterapi.VersionedAPISchema) (Factory, error) {
```

Comment:
should we now rename this to `NewChatCompletionTranslatorFactory` ?

Reply:
I've completely removed the Factory now, and moved the translator instantiation to the chat completion processor, as it does only make sense for that processor.

WDYT?

Reply:
sg